contracts/feirari/FuseGuardian.sol

```diff
@@ -0,0 +1,109 @@
+pragma solidity ^0.8.0;
+
+import "../refs/CoreRef.sol";
+import "../external/Unitroller.sol";
+
+/// @title a Fuse pause and borrow cap guardian used to expand access control to more Fei roles
+/// @author joeysantoro
+contract FuseGuardian is CoreRef {
+
+    /// @notice the fuse comptroller
+    Unitroller public immutable comptroller;
+
+    /// @param _core address of core contract
+    /// @param _comptroller the fuse comptroller
+    constructor(
+        address _core,
+        Unitroller _comptroller
+    ) CoreRef(_core) {
+        comptroller = _comptroller;
+        /// @notice The reason we are reusing the tribal chief admin role is it consolidates control in the OA,
+        /// and means we don't have to do another governance action to create this role in core
+        _setContractAdminRole(keccak256("TRIBAL_CHIEF_ADMIN_ROLE"));
+    }
+
+    // ************ BORROW GUARDIAN FUNCTIONS ************
+    /**
+      * @notice Set the given supply caps for the given cToken markets. Supplying that brings total underlying supply to or above supply cap will revert.
+      * @dev Admin or borrowCapGuardian function to set the supply caps. A supply cap of 0 corresponds to unlimited supplying.
+      * @param cTokens The addresses of the markets (tokens) to change the supply caps for
+      * @param newSupplyCaps The new supply cap values in underlying to be set. A value of 0 corresponds to unlimited supplying.
+      */
+    function _setMarketSupplyCaps(CToken[] memory cTokens, uint[] memory newSupplyCaps) public isGovernorOrGuardianOrAdmin {
+        comptroller._setMarketSupplyCaps(cTokens, newSupplyCaps);
+    }
+
+    function _setMarketSupplyCapsByUnderlying(address[] calldata underlyings, uint[] calldata newSupplyCaps) external {
+        _setMarketSupplyCaps(_underlyingToCTokens(underlyings), newSupplyCaps);
+    }
+
+    function _underlyingToCTokens(address[] memory underlyings) internal view returns (CToken[] memory) {
+        CToken[] memory cTokens = new CToken[](underlyings.length);
+        for (uint256 i = 0; i < underlyings.length; i++) {
```

Comment:
unchecked

Reply:
I'm probably going to leave as-is given this is an admin function and I'd rather optimize for readability than gas